addons/web_editor/static/src/js/editor/snippets.options.js

```diff
@@ -4278,32 +4285,43 @@ registry.ImageTools = SnippetOptionWidget.extend({
         new weWidgets.ImageCropWidget(this, this.$target[0]).appendTo(this.options.wysiwyg.$editable);
     },
     transform() {
-        if (this.$target.data('transfo-destroy')) {
-            this.$target.removeData('transfo-destroy');
-            return;
-        }
-        const document = this.$target[0].ownerDocument
+        const document = this.$target[0].ownerDocument;
         this.$target.transfo({ document });
+        const mouseup = this.updateUI.bind(this);
```

Comment:
I really don't like an option method being in charge of updating the UI. After an option method is called, the UI is automatically updated. I wonder if you should maybe just instantiate a Promise and resolve on the transform tools are done being used. Then the UI update would be done automatically.

Reply:
I am facing an issue with the promise resolution approach because if we want to resolve it when the tools are done being used, `_onUserValueUpdate` will display a loading effect until then. Should I start a small refactoring with a new `display-loading-effect` attribute on the widgets ?

Reply:
Mmh, yes maybe it's time to do something about that. However, what I would do is implement an event which can be triggered_up from an option method to disable the loading effect (then one to renable it). It makes more sense to do that from the options rather than from the widget as widgets are not supposed to know what they control: they just are elements which send values. The option method, however, has full control over what is actually done to the DOM and is the only one who can/should know what part of what it does should (or not) show a loading effect.